esprima.js

```diff
@@ -3955,6 +3956,12 @@
             }
         }
 
+        if (matchKeyword('let') || matchContextualKeyword('let')) {
```

Comment:
> Which test causes matchKeyword('let') to return true here?

In a case like this `esprima.parse('let x', { sourceType: 'module'})`. See function `advance()`, it automatically upgrades any strict mode reserved word (including `let`) to Keyword.